java/arcs/core/util/platform/jvm/ArcsStrictModeProvider.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.util
+
+/**
+ * Set [ArcsStrictMode] flags from the JVM environment.
+ */
+object ArcsStrictModeProvider {
+    fun strictHandles() = System.getProperty("ArcsStrictMode.strictHandles", "false") == "true"
```

Comment:
I feel like it'd be better to have this be explicitly configurable at runtime, rather than leaving it up to a system property? (e.g. within Application#onCreate for Android)

For inspiration: https://developer.android.com/reference/android/os/StrictMode

Reply:
The downside of not having a global flag is if you run to use TAP to audit all uses (that is, our TAP/GCB should turn this on for everything as a presubmit check), you have to hard-code it. 

I could add an API to make it explicitly configurable, but I feel like we should till have a global flag so we could turn it on during a TAP run.

Reply:
I'm not sure the ramifications of using System.getProperty on android are... which is why I think it's probably good to support both vectors.

Reply:
Also can this be a `val` instead?

Reply:
Changed to val. I put it as a method originally because I suspect it may have to be more complex later (e.g. WASM -> calling into WASI or JS to obtain environmental info)